frontend/packages/console-app/src/components/nodes/NodeTerminal.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import { Alert } from '@patternfly/react-core';
+import { Firehose, FirehoseResult, LoadingBox } from '@console/internal/components/utils';
+import { NodeKind, PodKind } from '@console/internal/module/k8s';
+import { PodExecLoader } from '../../../../../public/components/pod';
+import { ImageStreamTagModel, PodModel } from '../../../../../public/models';
+import { k8sCreate, k8sGet, k8sKillByName } from '../../../../../public/module/k8s';
+
+type NodeTerminalInnerProps = {
+  obj?: FirehoseResult<PodKind>;
+};
+
+type NodeTerminalProps = {
+  obj: NodeKind;
+};
+
+type NodeTerminalErrorProps = {
+  error: string;
+};
+
+// TODO: use https://github.com/openshift/console/blob/a5bc8fce01761043023a187e50baef1374b9013e/frontend/packages/dev-console/src/utils/shared-submit-utils.ts#L149-L154 instead
+const getRandomChars = () =>
+  Math.random()
+    .toString(36)
+    .substring(7);
+
+const getDebugImage = async (): Promise<string> => {
+  try {
+    const istag = await k8sGet(ImageStreamTagModel, 'tools:latest', 'openshift');
+    return istag.image.dockerImageReference;
+  } catch (e) {
+    return 'registry.redhat.io/rhel7/support-tools';
+  }
+};
+
+const getDebugPod = async (name: string, namespace: string, nodeName: string): Promise<PodKind> => {
+  const image = await getDebugImage();
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name,
+      namespace,
+      creationTimestamp: null,
+      annotations: {
+        'debug.openshift.io/source-container': 'container-00',
+        'debug.openshift.io/source-resource': `/v1, Resource=nodes/${nodeName}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [
+        {
+          name: 'host',
+          hostPath: {
+            path: '/',
+            type: 'Directory',
+          },
+        },
+      ],
+      containers: [
+        {
+          name: 'container-00',
+          image,
+          command: ['/bin/sh'],
+          resources: {},
+          volumeMounts: [
+            {
+              name: 'host',
+              mountPath: '/host',
+            },
+          ],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      nodeName,
+      hostNetwork: true,
+      hostPID: true,
+    },
+    status: {
```

Comment:
I'd remove the `status` stanza here

Reply:
the linter complains if there is no `status`.

Reply:
> the linter complains if there is no `status`.

We should fix the PodKind definition to make it optional.